doc/man/man1/ocpiav.1
.\"     Title: ocpiav
.\"    Author: [FIXME: author] [see http://www.docbook.org/tdg5/en/html/author]
.\" Generator: DocBook XSL Stylesheets vsnapshot <http://docbook.sf.net/>
.\"      Date: 09/04/2020
.\"    Manual: \ \&
.\"    Source: \ \&
.\"  Language: English
.\"
.TH "OCPIAV" "1" "09/04/2020" "\ \&" "\ \&"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
ocpiav \- command\-line tool for managing the OpenCPI AV GUI
.SH "SYNOPSIS"
.sp
\fBocpiav\fR [\fI<options>\fR] \fI<verb>\fR
.SH "DESCRIPTION"
.sp
The \fBocpiav(1)\fR command\-line tool is used to start the OpenCPI AV graphical user interface (GUI) for managing OpenCPI assets\&. It can also be used to set up a desktop shortcut for starting the AV tool from a CentOS7 development host\&.
.sp
The AV tool is a front end to \fBocpidev(1)\fR that can perform most, but not all, of the development tasks that \fBocpidev\fR can perform from within an Eclipse\-based GUI\&.
.SS "Installing AV"
.sp
AV is an optional OpenCPI development tool; the \fIOpenCPI Installation Guide\fR provides instructions on how to install it\&.
.SS "Using AV"
.sp
The \fIOpenCPI AV User Guide\fR describes the AV tool\(cqs menus and dialogs\&. The OpenCPI tutorials demonstrate how to use the AV tool to perform OpenCPI development tasks\&.
.SS "Working with OpenCPI Projects in AV"
.sp
AV is an Eclipse\-based tool and thus operates in a \(lqworkspace\(rq\&. The default workspace for the AV tool is located in the \fBav/workspace/\fR subdirectory of the OpenCPI installation\&. Do not create OpenCPI projects inside the AV tool because they will be created in the default AV workspace and will thus be inaccessible to other OpenCPI tools\&.
.sp
Use the \fBocpidev create project\fR operation to create a project outside of the AV tool and then use the AV tool to import it (do not copy it) into the AV workspace (File > Import > General > Import Existing Project dialog with the \(lqcopy\(rq option NOT checked\&.) The \fIOpenCPI User Guide\fR provides more information about this procedure, as do the OpenCPI tutorials\&.
.SH "VERBS"
.sp
The verbs supported by \fBocpiav\fR are:
.PP
\fBdesktop\fR
.RS 4
Add a selection for starting AV to the Applications > Programming menu bar on a CentOS7 development host\&.
.RE
.PP
\fBremove\fR
.RS 4
Remove the selection for AV from the Applications > Programming menu bar on a CentOS7 development host\&.
.RE
.PP
\fBrun\fR
.RS 4
Start AV from a
\fBbash\fR
shell\&.
.RE
.SH "NOUNS"
.sp
None\&.
.SH "OPTIONS"
.PP
\fB\-\-help\fR, \fB\-h\fR
.RS 4
Display usage information about the command\-line tool\&.
.RE
.SH "FILES/DIRECTORIES"
.sp
The following files and directories are relevant to OpenCPI AV operation:
.PP
\fBav\&.desktop\fR
.RS 4
The file that adds a menu selection for the OpenCPI AV tool to the Applications > Programming desktop environment menu bar on a CentOS7 development host\&. This file is created by the
\fBocpiav desktop\fR
operation in the CentOS7 user\(cqs home directory in the subdirectory
\fBlocal/share/applications/\fR\&.
.RE
.PP
\fBav/workspace/\fR
.RS 4
The subdirectory within the OpenCPI installation that contains the default Eclipse workspace for the AV tool\&.
.RE
.SH "EXAMPLES"
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Start the OpenCPI AV tool:
.sp
.if n \{\
.RS 4
.\}
.nf
ocpiav run
.fi
.if n \{\
.RE
.\}
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Add a menu selection for starting the OpenCPI AV tool from the CentOS7 desktop environment:
.sp
.if n \{\
.RS 4
.\}
.nf
ocpiav desktop
.fi
.if n \{\
.RE
.\}
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 3.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  3." 4.2
.\}
Remove the menu selection for the OpenCPI AV tool from the CentOS7 desktop environment:
.sp
.if n \{\
.RS 4
.\}
.nf
ocpiav remove
.fi
.if n \{\
.RE
.\}
.RE
.SH "BUGS"
.sp
See https://www\&.opencpi\&.org/report\-defects
.SH "RESOURCES"
.sp
See the main web site: https://www\&.opencpi\&.org
.SH "SEE ALSO"
.sp
ocpidev(1) ocpidev\-project(1) ocpidev\-registry(1)
.SH "COPYING"
.sp
Copyright (C) 2020 OpenCPI www\&.opencpi\&.org\&. OpenCPI is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version\&.


CHANGELOG.md
# [v2.0.0](https://gitlab.com/opencpi/opencpi/-/compare/v2.0.0-rc.2...v2.0.0) (2020-10-06)

Changes/additions since [OpenCPI Release v2.0.0-rc.2](https://gitlab.com/opencpi/opencpi/-/releases/v2.0.0-rc.2)

### New Features
- **tools**: add dockerfile for docker image used during GRCon20 workshops. (!386)(94e17772)

### Enhancements
- **app**: include the demo app from tutorial 1 in the tutorial project for completeness. (!373)(1566b9e9)
- **comp**: Add timed sample protocols. (!370)(90c31833)
- **comp**: add rcc and hdl workers for the tutorial 1 workers so we have both rcc and hdl of all of them. (!373)(1566b9e9)
- **comp**: cleanup another tutorial worker, `peak_detector.rcc`, for tutorial 2. (!373)(1566b9e9)
- **doc**: add files for the `timestamper_scdcd` datasheet to be compiled. (!357)(37931d3b)
- **runtime**: to be more consistent, enable "worker" attribute in app instances to include the authoring model suffix. (!373)(1566b9e9)
- **runtime**: allow "worker" variable in selection expressions. (!373)(1566b9e9)
- **runtime**: when there are no slaves indicated for a proxy instance (like from GRC), find slaves automatically if such a thing can be unambiguous.  This is still not fully inferred slaves, it is inferred slaves when they are already present, but not indicated as slaves. (!373)(1566b9e9)
- **runtime**: enable `OCPI_APPLICATION_PARAMS` to work better and for more options like `dump` and `verbose` (still not documented). (!373)(1566b9e9)
- **runtime**: have our hdl sim drivers kill sims on `SIGTERM` as well as `SIGINT`. (!373)(1566b9e9)
- **tools**: `ocpigen`: make `ocpi_debug` a hidden property. (!373)(1566b9e9)
- **tools**: `ocpigr37`: make `ocpi_container` a pseudo-property for GRC to be "partly hidden". (!373)(1566b9e9)
- **tools**: add platform, worker, model, and "done" pseudo-properties to the `ocpigr37` output, and put them last after normal properties. (!373)(1566b9e9)
- **tools**: `ocpigen`: make `ocpi_endian` a hidden property. (!373)(1566b9e9)
- **tools**: add env var `OCPI_DISTRO_BUILD` to enable generic x86_64 distro build feature. (!380)(e849452b)
- **tools**: build generic x86_64 GMP library when `OCPI_DISTRO_BUILD` env var is set. (!380)(e849452b)
- **tools**: add non-positional parameter handling to `ocpiadmin`. (!384)(cb1209c6)

### Bug Fixes
- **comp**: fix bug in `cdc_clk_gen.vhd`. (!378)(3ac9aac6)
- **hdl base**: fix an ancient bug in VHDL delta cycles that we previously blamed on being an isim-only bug. (!373)(1566b9e9)
- **runtime**: don't end up busy-waiting when the container has no applications to run at all. (!373)(1566b9e9)
- **tests**: fix typo in `ocpidev_directory_test.sh` relating to `ocpishow.py`. (!377)(0723c9aa)
- **tools**: fix an ancient bug in modelsim building that could report an error which then "disappeared" on rebuild. (!373)(1566b9e9)
- **tools**: make the tutorial project depend on platform so it can build for pluto and other OSPs. (!373)(1566b9e9)
- **tools**: fix `centos8-check.sh` script. (!375)(03a04a91)
- **tools**: fix `ocpishow.py` to have better error handling. (!377)(0723c9aa)
- **tools**: fix `project.py` (in `tools/python/_opencpi`) to have better error handling. (!377)(0723c9aa)

# [v2.0.0-rc.2](https://gitlab.com/opencpi/opencpi/-/compare/v2.0.0-rc.1...v2.0.0-rc.2) (2020-09-22)

Changes/additions since [OpenCPI Release v2.0.0-rc.1](https://gitlab.com/opencpi/opencpi/-/releases/v2.0.0-rc.1)

### New Features
- **tools**: add ability to specify host only prereqs. (!361)(3ac81b61)

### Enhancements
- **devops**: add script to automate creation of gitlab CI pipeline jobs for opencpi projects/platforms. (!351)(5b068b89)
- **doc**: add more man pages:  ocpihdl, ocpirun, opencpi, and some other updates. (!366)(207870c0)
- **tools**: add yaml-cpp v0.6.x as prereq (x86_64 only), removing weak dependency on boost. (!356)(be3e4cfa)

### Miscellaneous
- **comp**: add fifo_depth_p of 2048 to fifo.hdl build configuration. (!358)(fcd0e217)
- **tests**: suppress plutosdr from fir_complex_sse_ts.test due to lack of dsp resources. (!359)(387938a8)

# [v2.0.0-rc.1](https://gitlab.com/opencpi/opencpi/-/compare/v2.0.0-beta.1...v2.0.0-rc.1) (2020-08-28)

Changes/additions since OpenCPI Release v2.0.0-beta.1

### Enhancements
- **comp**: improve dc_offset_filter doc by adding transfer function. (!333)(364bb72e)
- **comp**: build socket_write.rcc for ubuntu18_04 platform. (!353)(7ef32634)
- **devops,tests**: remove obsolete ".bitz" files from git repo. (!337)(3d801f21)
- **doc**: add more man pages for OpenCPI commands. (!347)(6b7e41eb)
- **doc,tools**: add doc RPM to existing packaging machinery. (!329)(62cc8e74)
- **hdl base**: all hdl platforms have a standard test bitstream exported in a standard place. (!338)(e19fc095)
- **runtime,tools**: a proxy can have optional slaves so apps can have subsets, and the proxy can test or presence. (!338)(e19fc095)
- **tools**: enable python2 ACI in addition to python3 ACI. (!334)(3ea0db4d)
- **tools**: implement "-v" (verbose) option in "deploy-platform.sh" script. (!345)(5d7c5383)
- **tools**: add VERSION file that defines the version of OpenCPI. (!346)(b1e80c4d)
- **tools**: refactor `scripts/update-release.sh` to use new VERSION file. (!346)(b1e80c4d)

### Bug Fixes
- **comp**: `timestamper_scdcd.vhd`: add logic to confirm that EOF is valid. (!314)(b4918d13)
- **hdl base**: `complex_short_with_metadata_demarshaller.vhd`: add reset for EOF. (!314)(b4918d13)
- **hdl base**: `time_downsampler.vhd`: add reset for EOF. (!314)(b4918d13)
- **hdl base**: the ml605 now builds everything including the inactive and tutorial projects. (!338)(e19fc095)
- **runtime**: some property setting patterns, especially for remote device slaves, were broken and now fixed. (!344)(fb8e6c12)
- **tools**: fix export logic so as not to create broken symlinks. (!329)(62cc8e74)
- **tools**: build regressions on lesser used platforms not in normal pipeline. (!331)(a7f7cdd2)

tools/scripts/ocpiav.sh
#!/bin/bash --noprofile
# This file is protected by Copyright. Please refer to the COPYRIGHT file
# distributed with this source distribution.
#
# This file is part of OpenCPI <http://www.opencpi.org>
#
# OpenCPI is free software: you can redistribute it and/or modify it under the
# terms of the GNU Lesser General Public License as published by the Free
# Software Foundation, either version 3 of the License, or (at your option) any
# later version.
#
# OpenCPI is distributed in the hope that it will be useful, but WITHOUT ANY
# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
# A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
# details.
#
# You should have received a copy of the GNU Lesser General Public License along
# with this program. If not, see <http://www.gnu.org/licenses/>.
if [ -z "$OCPI_CDK_DIR" ]; then
  echo "The environment (specifically OCPI_CDK_DIR) is not set up."
  echo "You probably need to do \"source <wherever-the-cdk-is>/opencpi-setup.sh -s\"."
  exit 1
fi

usage () {
    cat <<-EOF
  Usage: ocpiav -h|--help|desktop|remove|run

    -h|--help   Show this message.

    desktop     Adds desktop environment "Applications" menu shortcut
                for running the AV GUI tool (creates "ocpiav.desktop"
                file in "~/.local/share/applications").

    remove      Removes the desktop environment menu shortcut
                (deletes "~/.local/share/applications/ocpiav.desktop").

    run         Runs the AV GUI tool.
EOF
    exit 1
}

[ -z "$1" -o "$1" = "-h" -o "$1" = "--help" ] && usage

file=~/.local/share/applications/ocpiav.desktop
case $1 in
  desktop)
    echo "Adding desktop environment menu item under \"Applications\" for running ocpiav tool."
    echo "This creates \"$file\"."
    mkdir -p ~/.local/share/applications
    sed "s-OCPI_CDK_DIR-$OCPI_CDK_DIR-" $OCPI_CDK_DIR/scripts/ocpiav.desktop > $file
    ;;
  remove)
    rm $file
    ;;
  run)
    if [ ! -x $OCPI_CDK_DIR/../av/eclipse/eclipse ]; then
	echo The OpenCPI AV GUI does not appear to be installed.
	exit 1
    fi
    exec $OCPI_CDK_DIR/../av/eclipse/eclipse> $OCPI_CDK_DIR/../av/av.log 2>&1 
    ;;
  *)
    echo "Illegal argument: $1" && usage
    ;;
esac
